dataeng/resources/tableau-config-backup.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+set -ex
+
+ssh -tt $USER_NAME@$TABLEAU_SERVER_IP -o StrictHostKeyChecking=no "S3_BUCKET=$S3_BUCKET" '
+
+# It gets the current date from system in 2022.04.05-10:35 format
+DATE="$(date '+%Y.%m.%d')"
+
+# It adds the date as prefix on backup name. Backup runs at 09:00 UTC so it also prefixed in the name.
+BACKUPNAME="${DATE}-tableau-config-backup.json"
+
+sudo runuser -l tsm_admin -c "tsm settings export -f /home/tableau_backups/tableau-config-backup.json"
+
+# Check if tableau backup file exist, exit if the file does not exist
+if [[ ! -f /home/tableau_backups/tableau-config-backup.json ]]
+then
+    echo "$BACKUPNAME Backup file does not exist. It seems to be some problem generating tableau backup. Refer to Administration of Tableau wiki docs to debug Tableau Regular Backup Automation"
+    exit 1
+else
+    echo "Copying backup file into s3"
+    aws s3 cp /home/tableau_backups/tableau-config-backup.json s3://$S3_BUCKET/$BACKUPNAME
```

Comment:
Rather than using s3 as an intermediary, what do you think about doing the following:

1. ssh into the box
2. if there is a backupfile present (at all) delete it
3. export the backup file with the tableau command
4. back on Jenkins, scp the file over

You are already using ssh, so you should be able to use the same credentials, and it removes the need to use a bucket. I don't know how useful this refactor would be. Just an idea.

Reply:
The reason I choose to copy the files on s3 was, it will keep the backup files for 60 days as we have life cycle configured for s3 bucket. So if we want to back track on the config files we will have config backups stored for last 60 days (1 backup each week = 8 backup files for last 8 weeks)

Reply:
Infact I realized the lifecycle rule was not attached on `config-backups/`. Created PR now: [https://github.com/edx/terraform/pull/4489](https://github.com/edx/terraform/pull/4489)